src/identity/json.rs

```diff
@@ -0,0 +1,57 @@
+use crate::identity::AccessResponse;
+use crate::identity::IdentityStore;
+use crate::identity::Outcome;
+use crate::util::Result;
+use std::path::Path;
+use std::fs::File;
+
+#[derive(Deserialize)]
+pub struct JsonSettings {
+    pub file_path: String,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct User {
+    username: String,
+    token: String,
+    access: bool
+}
+
+pub struct Json {
+    settings: JsonSettings,
+}
+
+impl Json {
+    pub fn new(settings: JsonSettings) -> Self {
+        Self { settings }
+    }
+}
+
+impl IdentityStore for Json {
+    fn access(&mut self, token: &str) -> Result<AccessResponse> {
+        let json_file_path = Path::new(&self.settings.file_path);
+        let file = File::open(json_file_path).expect("json identity file not found");
```

Comment:
`expect` panics on an error. Instead, simply return the error by using the question mark (`?`) operator.

Reply:
ok